pandas/tests/series/test_combine_concat.py

```diff
@@ -116,8 +116,35 @@ def test_update(self):
         df = DataFrame([{"a": 1}, {"a": 3, "b": 2}])
         df['c'] = np.nan
 
-        # this will fail as long as series is a sub-class of ndarray
-        # df['c'].update(Series(['foo'],index=[0])) #####
+        df['c'].update(Series(['foo'], index=[0]))
+        expected = DataFrame([[1, np.nan, 'foo'], [3, 2., np.nan]],
+                             columns=['a', 'b', 'c'])
+        assert_frame_equal(df, expected)
+
+    def test_update_dtypes(self):
+        s = Series([1., 2., False, True])
```

Comment:
can you either parameterize this or copy s each time from the original so its clear what is changing.

Reply:
@jreback : Looking this over now, I'm leaning towards parameterization.  Looks cleaner that way (and no need for all of this copying).

@h-vetinari : WDYT?